core/client/fs/src/main/java/alluxio/client/file/options/FileSystemOptions.java

```diff
@@ -108,10 +164,12 @@ public Builder setDataCacheEnabled(boolean dataCacheEnabled) {
     }
 
     /**
-     * @return the worker net address
+     * @return the file system options
      */
     public FileSystemOptions build() {
-      return new FileSystemOptions(mMetadataCacheEnabled, mDataCacheEnabled);
+      return new FileSystemOptions(mFileSystemType, mMetadataCacheEnabled, mDataCacheEnabled,
+          mUfsFileSystemOptions == null
+              ? new UfsFileSystemOptions.Builder().build() : mUfsFileSystemOptions);
```

Comment:
I think the more canonical way of builder pattern is to pass the builder object into the constructor, because in that way, you don't need to modify the constructor parameters nor the caller function like here when you add more parameters.
More specifically, if nothing is going to change for the `FileSystemOptions`, it's taking only one parameter, so a Builder pattern is actually not necessary. If one needs to add more parameters to `FileSystemOptions`, not only they need to add the parameters (along with the setter method) in the Builder, but also need to modify the build() method, and the constructor of `FileSystemOptions`. Whereas if we pass the Builder object into the `FileSystemOptions` constructor and we want to add more parameters, we only need to add the parameter (along with getter and setter) in the Builder and make `FileSystemOptions` take the parameter from the builder object. Plus we do one less check (no need to check if `mUfsFileSystemOptions` is null or not here. Can be taken care of in the constructor)

Reply:
Same for the `FuseOptions`